crates/rethnet_eth/src/remote.rs

```diff
@@ -166,3 +166,28 @@ impl serde::Serialize for ZeroXPrefixedBytes {
         serializer.serialize_str(&format!("0x{}", hex::encode(&self.inner)))
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    fn help_test_block_spec_serde(block_spec: BlockSpec) {
+        let json = serde_json::json!(block_spec).to_string();
+        let block_spec_decoded: BlockSpec = serde_json::from_str(&json)
+            .unwrap_or_else(|_| panic!("should have successfully deserialized json {json}"));
+        assert_eq!(block_spec, block_spec_decoded);
+    }
+
+    #[test]
+    fn test_serde_block_spec() {
+        help_test_block_spec_serde(BlockSpec::Number(U256::from(123)));
+        help_test_block_spec_serde(BlockSpec::latest());
```

Comment:
It's still not complete w.r.t to the `enum BlockTag`

Reply:
addressed in https://github.com/NomicFoundation/hardhat/pull/4132/commits/fc75935c567aff233ff709c21ed264fe2ecb3444